### PR TITLE
Phpstan Level6 for View.php

### PR DESCRIPTION
**Pull request by ColonelMoutarde**
Changes proposed in this pull request:

- Phpstan Level6 for View.php

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested

### CODE REVIEW

File: tests/phpstan-next.txt
Comment (Alkarex) [inline tests/phpstan-next.txt:1]: 👍🏻

File: lib/Minz/View.php [hunk 1]
```diff
@@ -8,24 +8,28 @@
  * The Minz_View represents a view in the MVC paradigm
  */
 class Minz_View {
-	const VIEWS_PATH_NAME = '/views';
-	const LAYOUT_PATH_NAME = '/layout/';
-	const LAYOUT_DEFAULT = 'layout';
-
+	public const VIEWS_PATH_NAME = '/views';
+	public const LAYOUT_PATH_NAME = '/layout/';
+	public const LAYOUT_DEFAULT = 'layout';
```
Comment (Alkarex) [inline lib/Minz/View.php]: `private` should be enough?

File: lib/Minz/View.php [hunk 2]
```diff
@@ -95,57 +99,61 @@ private function includeFile($filename) {
...
 	}
 
 	/**
-	 * Ajoute un élément du layout
-	 * @param string $part l'élément partial à ajouter
+	 * Adds a layout element
+	 * @param string $part the partial element to be added
 	 */
-	public function partial ($part) {
+	public function partial (string $part): void {
```
Comment (Alkarex) [inline lib/Minz/View.php]:
```suggestion
	public function partial(string $part): void {
```

File: lib/Minz/View.php [hunk 3]
```diff
@@ -95,57 +99,61 @@ private function includeFile($filename) {
...
 	}
 
 	/**
-	 * Affiche un élément graphique situé dans APP./views/helpers/
-	 * @param string $helper l'élément à afficher
+	 * Displays a graphic element located in APP./views/helpers/
+	 * @param string $helper the element to be displayed
 	 */
-	public function renderHelper ($helper) {
+	public function renderHelper (string $helper): void {
```
Comment (Alkarex) [inline lib/Minz/View.php]:
```suggestion
	public function renderHelper(string $helper): void {
```

**Comment by Alkarex** [comment]
I have made a few more improvements / fixes https://github.com/FreshRSS/FreshRSS/pull/5269/commits/c0ec012e2cd2d8b8edc57e82668b9b8dfe60464e
Please double-check.
If you feel like it, it would be nice to address `./app/Models/View.php`, which is the natural next step after `./lib/Minz/View.php`, either in this PR or a new one.

**Comment by ColonelMoutarde** [comment]
> I have made a few more improvements / fixes [c0ec012](https://github.com/FreshRSS/FreshRSS/commit/c0ec012e2cd2d8b8edc57e82668b9b8dfe60464e) Please double-check. If you feel like it, it would be nice to address `./app/Models/View.php`, which is the natural next step after `./lib/Minz/View.php`, either in this PR or a new one.

I add ./app/Models/View.php quickly

**Comment by ColonelMoutarde** [comment]
> I have made a few more improvements / fixes [c0ec012](https://github.com/FreshRSS/FreshRSS/commit/c0ec012e2cd2d8b8edc57e82668b9b8dfe60464e) Please double-check. If you feel like it, it would be nice to address `./app/Models/View.php`, which is the natural next step after `./lib/Minz/View.php`, either in this PR or a new one.

done

**Comment by Alkarex** [comment]
To get some of types right, some more types had to be added to related files.
I have made a number of fixes and improvements. Please double-check if possible

**Comment by ColonelMoutarde** [comment]
> To get some of types right, some more types had to be added to related files. I have made a number of fixes and improvements. Please double-check if possible

I think it's Ok

File: app/views/index/logs.phtml [hunk 4]
```diff
@@ -14,7 +14,7 @@
 	?>
 
 	<?php if (!empty($items)) { ?>
-	<?php $this->logsPaginator->render('logs_pagination.phtml', 'page'); ?>
+	<?php $this->logsPaginator->render('logs_pagination.phtml', 0); ?>
```
Comment (Alkarex) [inline app/views/index/logs.phtml:17]: This was wrong, breaking the pagination. Fixed in https://github.com/FreshRSS/FreshRSS/pull/5403

File: app/Models/CategoryDAO.php [hunk 5]
```diff
@@ -435,13 +436,12 @@ public static function countUnread(array $categories, int $minPriority = 0): int
 		return $n;
 	}
 
-	public static function daoToCategoryPrepopulated($listDAO) {
+	/**
+	 * @param array<string,mixed> $listDAO
+	 * @return array<int,FreshRSS_Category>
+	 */
+	private static function daoToCategoryPrepopulated(array $listDAO) {
```
Comment (Alkarex) [inline app/Models/CategoryDAO.php:443]: Native array return type forgotten: fixed in https://github.com/FreshRSS/FreshRSS/pull/6046

**Pull request merged**